include/deal.II/cgal/surface_mesh.h

```diff
@@ -25,12 +25,40 @@
 #include <deal.II/cgal/utilities.h>
 
 #ifdef DEAL_II_WITH_CGAL
+#  include <CGAL/Polygon_mesh_processing/corefinement.h>
+#  include <CGAL/Polygon_mesh_processing/triangulate_faces.h>
 #  include <CGAL/Surface_mesh.h>
+#  include <CGAL/Triangulation_3.h>
+#  include <CGAL/convex_hull_3.h>
+
 
 DEAL_II_NAMESPACE_OPEN
 
 namespace CGALWrappers
 {
+  /**
+   * The following functions are built on top of
+   * [CGAL::Surface_mesh](https://doc.cgal.org/latest/Surface_mesh/index.html)
+   * and the `CGAL::PolygonMeshProcessing` namespace. With these objects we can
+   * perform boolean operations on deal.II types such as difference, union and
+   * intersection and many more. They allow to generate quadrature rules to
+   * integrate integrate functions over arbitrarily shaped polygons using
+   * `collect_quadratures_inside_surface()`.
+   *
+   * The shaded region in the following picture is the intersection between a
+   * cube and a tetrahedron, converted to a deal.II triangulation
+   *
+   * @image html intersection_cube_with_tetrahedron.png
+   *
```

Comment:
Perfect!

Can you also add the same picture computin the union and the difference?